src/Analyzer.Utilities/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -93,8 +93,18 @@ public static IEnumerable<ITypeSymbol> GetBaseTypesAndThis(this ITypeSymbol type
             }
         }
 
-        public static bool DerivesFrom(this INamedTypeSymbol symbol, INamedTypeSymbol candidateBaseType)
+        public static bool DerivesFrom(this ITypeSymbol symbol, ITypeSymbol candidateBaseType, bool baseTypesOnly = false)
```

Comment:
Yes, I understand that. However, in this case, if you look at the definition of `ITypeSymbol.AllInterfaces`
```cs
 ImmutableArray<INamedTypeSymbol> AllInterfaces { get; }
```
It is actually returns an array of `INamedTypeSymbol`, therefore I believe this change only provides a false sense of wider applicability with overhead from unnecessarily casting entire array. 

I'm not sure if this would cause any problem at runtime if you tries to pass arguments of type `ITypeSymbol` that's not an `INamedTypeSymbol`. Maybe @srivatsn @lgolding knows.